syntax = "proto3";

package bank;

service CurrencyService {
    rpc getCurrency (CurrencyRequest) returns (stream CurrencyReply) {
    }
}

message CurrencyRequest {
    repeated Currency currency = 1;
}

message CurrencyReply {
    repeated CurrencyTable currencyTable = 1;
}

message CurrencyTable {
    Currency currency = 1;
    float value = 2;
}

enum Currency {
    PLN = 0;
    USD = 1;
    EURO = 2;
}